integration/store_gateway_limits_hit_test.go

```diff
@@ -0,0 +1,225 @@
+)
+
+const (
+	blockRangePeriod = 5 * time.Second
+	ingesterTag      = "ingester"
+	distributorTag   = "distributor"
+	storeGatewayTag  = "store-gateway"
+	querierTag       = "querier"
+	orgID            = "test"
+	series1          = "series_1"
+	series2          = "series_2"
+	series3          = "series_3"
+	series4          = "series_4"
+)
+
+func Test_MaxSeriesPerQueryLimitHit(t *testing.T) {
+	tests := map[string]struct {
+		additionalStoreGatewayFlags map[string]string
+		additionalQuerierFlags      map[string]string
+		expectedErrorKey            string
+	}{
+		"when store-gateway hits max_fetched_series_per_query, 'exceeded series limit' is returned": {
+			additionalStoreGatewayFlags: map[string]string{"-querier.max-fetched-series-per-query": "3"},
+			expectedErrorKey:            storegateway.ErrSeriesLimitMessage,
+		},
+		"when querier hits max_fetched_series_per_query, 'err-mimir-max-series-per-query' is returned": {
+			additionalQuerierFlags: map[string]string{"-querier.max-fetched-series-per-query": "3"},
+			expectedErrorKey:       string(globalerror.MaxSeriesPerQuery),
+		},
+	}
+
+	for testName, testData := range tests {
+		t.Run(testName, func(t *testing.T) {
+			scenario, mimirServices, client := createClient(t, testData.additionalStoreGatewayFlags, testData.additionalQuerierFlags)
+			defer scenario.Close()
+
+			timeStamps12, timeSeries12 := createTimeSeries(t, []string{series1, series2})
+			pushTimeSeries(t, client, timeSeries12[series1])
+			pushTimeSeries(t, client, timeSeries12[series2])
+
+			// ensures that data will be queried from store-gateway, and not from ingester: at this stage 1 block of data should be stored
+			waitUntilShippedToStorage(t, mimirServices, 1)
+
+			// Verify we can successfully read the data we have just pushed
+			rangeResultResponse, _, err := client.QueryRangeRaw("{__name__=~\"series_.+\"}", timeStamps12[series1], timeStamps12[series2].Add(1*time.Hour), time.Second)
+			require.NoError(t, err)
+			require.Equal(t, http.StatusOK, rangeResultResponse.StatusCode)
+
+			timeStamps34, timeSeries34 := createTimeSeries(t, []string{series3, series4})
+			pushTimeSeries(t, client, timeSeries34[series3])
+			pushTimeSeries(t, client, timeSeries34[series4])
+
+			// ensures that data will be queried from store-gateway, and not from ingester: at this stage 2 blocks of data should be stored
+			waitUntilShippedToStorage(t, mimirServices, 2)
+
+			// Verify we cannot read the data we just pushed because the limit is hit, and the status code 422 is returned
+			rangeResultResponse, rangeResultBody, err := client.QueryRangeRaw("{__name__=~\"series_.+\"}", timeStamps12[series1], timeStamps34[series4].Add(1*time.Hour), time.Second)
+			require.NoError(t, err)
+			require.Equal(t, http.StatusUnprocessableEntity, rangeResultResponse.StatusCode)
+			require.True(t, strings.Contains(string(rangeResultBody), testData.expectedErrorKey))
+		})
+	}
+}
+
+func Test_MaxChunksPerQueryLimitHit(t *testing.T) {
```

Comment:
[nit] I think you can merge the two tests together. Other comments I left assume we do merge the two tests.